crates/metadata-server/src/grpc/client.rs

```diff
@@ -497,28 +496,32 @@ impl ChannelWithAddress {
 #[derive(Debug)]
 struct Channels {
     initial_channels: Vec<ChannelWithAddress>,
-    channels: HashMap<PlainNodeId, ChannelWithAddress>,
+    channels: IndexMap<PlainNodeId, ChannelWithAddress>,
+    channel_index: usize,
 }
 
 impl Channels {
     fn new(initial_channels: Vec<ChannelWithAddress>) -> Self {
+        assert!(!initial_channels.is_empty());
         Channels {
             initial_channels,
-            channels: HashMap::default(),
+            channels: IndexMap::default(),
+            channel_index: 0,
         }
     }
 
     fn register(&mut self, plain_node_id: PlainNodeId, channel: ChannelWithAddress) {
         self.channels.insert(plain_node_id, channel);
     }
 
-    fn choose_random(&self) -> Option<ChannelWithAddress> {
-        let mut rng = rand::rng();
+    fn choose_next_round_robin(&mut self) -> Option<ChannelWithAddress> {
+        self.channel_index =
+            (self.channel_index + 1) % (self.channels.len() + self.initial_channels.len());
         let chosen_channel = self
             .channels
             .values()
             .chain(self.initial_channels.iter())
-            .choose(&mut rng)
+            .nth(self.channel_index)
```

Comment:
Instead of `nth` we could directly access `self.channels.get_index(self.channel_index)` if `channel_index < self.channels.len()` and else `self.initial_channels[self.channel_index - self.channels.len()]` to avoid the O(n) access (although `n` is probably quite small here.